components/console-backend-service/internal/gqlschema/schema.graphql

```diff
@@ -1014,6 +1014,82 @@ type VersionInfo {
     kymaVersion: String!
 }
 
+# Trigger
+
+type Trigger {
+    name: String!
+    namespace: String!
+    broker: String!
+    filterAttributes: JSON!
+    subscriber: Subscriber!
+    status: TriggerStatus!
+}
+
+type TriggerStatus {
+    reason: [String!]
+    status: TriggerStatusType!
+}
+
+enum TriggerStatusType {
+    FAILED
+    READY
+}
+
+type Subscriber {
+    uri: String
+    ref: SubscriberRef
+}
+
+type SubscriberRef {
+    apiVersion: String!
+    kind: String!
+    name: String!
+    namespace: String!
+}
+
+input SubscriberInput {
+    uri: String
+    ref: SubscriberRefInput
+}
+
+input SubscriberRefInput {
+    apiVersion: String!
+    kind: String!
+    name: String!
+    namespace: String!
+}
+
+input TriggerCreateInput {
+    name: String!
+    namespace: String!
+    broker: String!
+    filterAttributes: JSON!
+    subscriber: SubscriberInput!
+}
+
+input TriggerMetadataInput {
+    name: String!
+    namespace: String!
+}
+
+type TriggerMetadata {
+    name: String!
+    namespace: String!
+}
+
+input OwnerReference {
+    apiVersion: String!
+    blockOwnerDeletion: Boolean!
+    controller: Boolean!
```

Comment:
`controller` and `blockOwnerDeletion` are necessary? Completing those fields with information from UI can be difficult. Please do they as optional.